packages/utils/src/eslint-utils/rule-tester/dependencyConstraints.ts

```diff
@@ -0,0 +1,77 @@
+import * as semver from 'semver';
+
+interface SemverVersionConstraint {
+  readonly type: 'semver';
+  readonly range: string;
+  readonly options?: Parameters<typeof semver.satisfies>[2];
+}
+interface AtLeastVersionConstraint {
+  readonly type: 'at-least';
+  readonly version:
+    | `${number}`
+    | `${number}.${number}`
+    | `${number}.${number}.${number}`
+    | `${number}.${number}.${number}-${string}`;
+}
+type VersionConstraint = SemverVersionConstraint | AtLeastVersionConstraint;
+interface DependencyConstraint {
+  /**
+   * Passing a string for the value is shorthand for the 'at-least' constraint
+   */
+  readonly [packageName: string]:
+    | VersionConstraint
+    | AtLeastVersionConstraint['version'];
+}
+
+const BASE_SATISFIES_OPTIONS: semver.RangeOptions = {
+  includePrerelease: true,
+};
+
+function satisfiesDependencyConstraint(
+  packageName: string,
+  constraintIn: DependencyConstraint[string],
+): boolean {
+  const constraint: VersionConstraint =
+    typeof constraintIn === 'string'
+      ? {
+          type: 'at-least',
+          version: constraintIn,
+        }
+      : constraintIn;
+
+  const satisfiesArguments: [string, SemverVersionConstraint['options']] =
+    constraint.type === 'at-least'
+      ? [`>=${constraint.version}`, BASE_SATISFIES_OPTIONS]
```

Comment:
🤔 I'm not convinced it's worth the extra code & API surface area to have `{ type: 'at-least' }` as an option. Users can get the same behavior quickly from the `string` option or with just a little more effort use the `{ type: 'semver' }`.

Proposal: how about removing `'at-least'` and the `type` field altogether? That way the shape would look like:

```ts
interface SemverVersionConstraint {
  readonly range: string;
  readonly options?: Parameters<typeof semver.satisfies>[2];
}

type VersionConstraint = string | SemverVersionConstraint;
```

Reply:
Yeah part of me was building a surface that was future-proofed in case we added more. But it's probably over-engineering and not worth the effort.